ipykernel/kernelapp.py

```diff
@@ -142,9 +143,9 @@ class IPKernelApp(BaseIPythonApplication, InteractiveShellApp, ConnectionFileMix
     debug_shell_socket = Any()
     stdin_socket = Any()
     iopub_socket = Any()
-    iopub_thread = Any()
-    control_thread = Any()
-    shell_channel_thread = Any()
+    iopub_thread: Thread
+    control_thread: Thread
+    shell_channel_thread: Thread
```

Comment:
[BaseThread](https://github.com/ipython/ipykernel/blob/3089438d84447bf859556b89f528ef59332803c5/ipykernel/thread.py#L11)?

Reply:
I'm not sure if it's overridden in subclasses somewhere, but sure I can try so if mypy ever complains.